vision.py

```diff
@@ -9,16 +12,69 @@
 from loguru import logger
 from tqdm import tqdm
 from PIL import Image
-import tracking
+import moviepy.video.io.ImageSequenceClip
+from rq import get_current_job
+from flask_socketio import SocketIO
 
+import tracking 
 
-def get_tracking_video(fpath_video):
-    logger.info(f'get_tracking_video fpath_video: {fpath_video}')
+def display_message(status, job, socketio):
+    '''
+    Given a status message, a job, and an open socket, updates the UI
+    to include the status message
+    Parameters:
+    - status: string
+    - job: REDIS job
+    - socketio: Socketio object
+    '''
+    socketio.emit('progress display', 
+                 {'status': status, 'id': str(job.id)}, json=True)
+    job.meta['status'] = status
+    job.save()
+    logger.info('Changing status to "{}"'.format(status))
+
+def get_tracking_video(fpath_video, output_file):
+    '''
+    Given the path to a video, creates a tracking video and writes it to the
+    path output_file
+    Parameters:
+    - str fpath_video: path to video file
+    - str output_file: path to output file (doesn't need to already exist)
+    '''
+    logger.info('Cuda available {}'.format(torch.cuda.is_available()))
+    job = get_current_job()
+
+    # Opening socket
+    socketio = SocketIO(
+        message_queue=os.getenv('REDIS_URL'), 
+        cors_allowed_origins='*'
+    )
+    
+    # Getting images from video
+    display_message('Getting images from video', job, socketio)
+    video, extension = fpath_video.split('.')
     image_gen  = _get_images_from_video(fpath_video)
     images = [image for image in image_gen]
+    
+    # Getting predictions from images
+    display_message('Getting predictions from images', job, socketio)
     predictions = _get_predictions_from_images(images)
+    
+    # Getting tracks from predictions
+    display_message('Getting tracks from predictions', job, socketio)
     tracks = tracking.get_tracks(predictions)
-    fpath_tracking_video = _get_video_from_tracks(tracks, images)
+
+    # Making video from tracks
+    display_message('Making video from tracks', job, socketio)
+    fpath_tracking_video = _get_video_from_tracks(tracks, images, output_file)
+    
+    # Done
+    socketio.emit('progress display', {'status': 'Done', 'id': str(job.id), 
+                 'fname': fpath_tracking_video}, json=True)      
```

Comment:
abrichr 9 days ago Member
Formatting as above

Reply:
created dictionary variable and passed that instead of supplying the dictionary directly